From b21df1e6c7b6a31d93b3a6a663b0a7a56ca9b160 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 26 Aug 2014 14:11:19 -0400
Subject: [ethernet] igb: Add additional get_phy_id call for i354 devices

Message-id: <1409062359-13063-2-git-send-email-sassmann@redhat.com>
Patchwork-id: 89122
O-Subject: [RHEL7.1 PATCH 01/81] igb: Add additional get_phy_id call for i354 devices
Bugzilla: 1091121
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Neil Horman <nhorman@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Carolyn Wyborny <carolyn.wyborny@intel.com>

This patch fixes a problem where some ports can fail to initialize on a
cold boot. This patch adds an additional call to read the PHY id for i354
devices in order workaround the hardware problem.

Signed-off-by: Carolyn Wyborny <carolyn.wyborny@intel.com>
Tested-by: Aaron Brown <aaron.f.brown@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit bb1d18d1ad6e216535b994d68097ae107f4f35fb)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/igb/e1000_82575.c b/drivers/net/ethernet/intel/igb/e1000_82575.c
index 79b5835..47c2d10 100644
--- a/drivers/net/ethernet/intel/igb/e1000_82575.c
+++ b/drivers/net/ethernet/intel/igb/e1000_82575.c
@@ -719,6 +719,10 @@ static s32 igb_get_phy_id_82575(struct e1000_hw *hw)
  u32 ctrl_ext;
  u32 mdic;
 
+ /* Extra read required for some PHY's on i354 */
+ if (hw->mac.type == e1000_i354)
+  igb_get_phy_id(hw);
+
  /* For SGMII PHYs, we try the list of possible addresses until
   * we find one that works.  For non-SGMII PHYs
   * (e.g. integrated copper PHYs), an address of 1 should
-- 
1.7.1